Sources/KituraNet/HTTP/HTTPServerRequest.swift

```diff
@@ -67,22 +69,25 @@ public class HTTPServerRequest: ServerRequest {
         }
 
         if let hostname = headers["Host"]?.first {
-            url.append(hostname)
-            if !hostname.contains(":") {
-                url.append(":")
-                url.append(String(describing: localAddressPort))
-            }
+            _urlComponents?.host = hostname
         } else {
             Log.error("Host header not received")
             let hostname = localAddress
-            url.append(hostname == "127.0.0.1" ? "localhost" : hostname)
-            url.append(":")
-            url.append(String(describing: localAddressPort))
+            _urlComponents?.host = hostname == "127.0.0.1" ? "localhost" : hostname
+        }
+
+        _urlComponents?.port = Int(localAddressPort)
+
+        let uriComponents = _urlString.components(separatedBy: "?")
```

Comment:
It may be better to use `String.split()` here. See https://github.com/IBM-Swift/Kitura-net/pull/287